llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1678,7 +1678,9 @@ void SPIRVToLLVM::addStoreInstRecursively(SPIRVType *const spvType, Value *store
   if (isTypeWithPadRowMajorMatrix(storeType)) {
     auto storePair = createLaunderRowMajorMatrix(storeType, storePointer);
     storePointer = storePair.second;
-    storeType = storePointer->getType()->getPointerElementType();
+    storeType = storePair.first;
+    // TODO: Remove this when LLPC will switch fully to opaque pointers.
+    assert(storeType == storePointer->getType()->getNonOpaquePointerElementType());
```

Comment:
`cast<PointerType>(storePointer->getType())->isOpaqueOrPointeeTypeMatches(storeType)` should work with and without opaque pointers (here and below).

Reply:
Yes, you are right. We could use it but do we want to have any checks after the transition to opaque pointers ? When opaque pointers will be "turned on" then `isOpaqueOrPointeeTypeMatches` will return always true.

Reply:
Right, I assumed we want to be able to run with or without opaque pointers for some weeks when opaque pointers are almost working?
That makes it easier to switch between the modes, one can just add `-opaque-pointers` when running amdllpc.

Reply:
OK, this is a good point !
I will update review.